### PR TITLE
Include old preview versions

### PR DESCRIPTION
**Pull request by danielrbradley**
- Include latest preview version if no new stable versions after 1 year (adds 27 resources).
- Consider all version when calculating additions. Only ignore private preview versions. This takes us to up 242 additional resources in the default version.
- Fix sorting of versions so that previews are considered older than stable versions. 
- Upgrade CI to Go 1.20 so we can use `time.Compare`.

Fixes https://github.com/pulumi/pulumi-azure-native/issues/2359
Fixes https://github.com/pulumi/pulumi-azure-native/issues/2477

### CODE REVIEW

**Review by thomas11 — APPROVED** [review]
LGTM although I'm not 100% sure I totally understand the third commit.

File: provider/pkg/versioning/defaultVersion.go [hunk 1]
```diff
@@ -294,12 +303,33 @@ func filterCandidateVersions(versions VersionResources, previewPolicy string) []
 		// If no more to iterate through, just add
 		if i == 0 || !containsRecentStable(orderedVersions[:i], version) {
 			candidateVersions.Add(version)
+			continue
+		}
+
+		isOld, err := isMoreThanOneYearOld(version)
```
Comment (mikhailshilkov) [inline provider/pkg/versioning/defaultVersion.go:309]: Could you please explain the genesis of this 1-year rule?

File: provider/pkg/versioning/defaultVersion.go [hunk 2]
```diff
@@ -181,8 +181,16 @@ func buildSpec(providerName string, versions VersionResources, curations Curatio
 	if existing.Additions != nil {
 		existingAdditions = *existing.Additions
 	}
+	sortedVersions := []string{}
+	for version := range versions {
+		sortedVersions = append(sortedVersions, version)
+	}
+	sort.Strings(sortedVersions)
```
Comment (mikhailshilkov) [inline provider/pkg/versioning/defaultVersion.go:188]: Note that alphabetical sorting will put stable versions before preview ones (v20230101 < v20230101preview).

**Review by mikhailshilkov — COMMENTED** [review]
Is schema-tools compare not wired correctly? It shows a bunch of breaking changes and also only 31 new resources...

File: provider/pkg/versioning/defaultVersion.go (hunk 1 above)
Comment (danielrbradley) [reply provider/pkg/versioning/defaultVersion.go:309]: It's a somewhat arbitary amount of time, but there's a pattern of a preview version being replaced with a stable version after a few months - at which point we'd prefer to be pulling from stable versions if possible.

File: provider/pkg/versioning/defaultVersion.go (hunk 2 above)
Comment (danielrbradley) [reply provider/pkg/versioning/defaultVersion.go:188]: Yes, good catch. I'll fix this up

**Pull request merged**